runtime/connection.cpp

```diff
@@ -315,16 +315,16 @@ void Connection::start()
 void Connection::stop()
 {
 	scout_debug << "Connection::stop()";
-	_writer.stop();
-	_writer.join();
-	_reader.stop();
-	_reader.join();
 	try {
 		_reader.socket()->shutdownReceive();
```

Comment:
i know we talked about it, but i already forgot.
in recent fix8 they changed this shutdownReceive() to just shutdown() is this something we want?

Reply:
It's conventional to shut down write, this tells the server to not accept further requests.  The client is free to read off any pending responses before stopping.  That's what should happen in this case.
If we call shutdown for read and write, it's possible to receive a partial final message if it's fragmented, causing possible downstream errors.

Reply:
receiving partial final message is fine,  because fix8 is capable of getting partial messages, it always waits for checksum and only then builds up a message.

Reply:
Given we're talking about the reader, and it seems they just want to cut the connection, shutdown(read/write) is ok, but seems a little desperate.  I'll change it to shutdown() as that's what they've gone with in the end.